libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -642,9 +642,24 @@ void CodeTransform::operator()(ForLoop const& _forLoop)
 	m_assembly.appendLabel(loopEnd);
 
 	finalizeBlock(_forLoop.pre, stackStartHeight);
+	m_context->forLoopStack.pop();
 	m_scope = originalScope;
 }
 
+void CodeTransform::operator()(Break const& _break)
+{
+	yulAssert(!m_context->forLoopStack.empty(), "Invalid break-statement. Requires surrounding for-loop in code generation.");
+	m_assembly.setSourceLocation(_break.location);
+	m_assembly.appendLabel(m_context->forLoopStack.top().done);
```

Comment:
`appendLabel` is not the right function. You need `appendJumpTo` and probable a bunch of POP operations before that.